opslevel/resources_opslevel_team_tag.go

```diff
@@ -0,0 +1,157 @@
+package opslevel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceTeamTag() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a team tag",
+		Create:      wrap(resourceTeamTagCreate),
+		Read:        wrap(resourceTeamTagRead),
+		Update:      wrap(resourceTeamTagUpdate),
+		Delete:      wrap(resourceTeamTagDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"team": {
+				Type:        schema.TypeString,
+				Description: "The id of the team that this will be added to.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"team_alias": {
+				Type:        schema.TypeString,
+				Description: "The alias of the team that this will be added to.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"key": {
+				Type:         schema.TypeString,
+				Description:  "The tag's key.",
+				ForceNew:     false,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(TagKeyRegex, TagKeyErrorMsg),
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Description: "The tag's value.",
+				ForceNew:    false,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceTeamTagCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	team, err := findTeam("team_alias", "team", d, client)
+	if err != nil {
+		return err
+	}
+
+	input := opslevel.TagCreateInput{
+		Id: team.Id,
+
+		Key:   d.Get("key").(string),
+		Value: d.Get("value").(string),
+	}
+	resource, err := client.CreateTag(input)
```

Comment:
Is `CreateTag` the correct thing here?  Should we be using `AssignTag` ? (i'm honestly not sure)

Reply:
I actually wasn't even aware that the tagAssign mutation existed until just now to be honest (it still works as expected with the new team tag type since they share the same TaggableResource type used as input).

I was following what we did in `resources_opslevel_service_tag.go` since these are basically identical (so if we change this I think we should also do the same there). I think the nice thing about `AssignTag ` is that we could specify multiple tags to assign at once, but I think `CreateTag` is more in line with a function named `resourceTeamTagCreate` (I notice our other resources in our terraform provider typically only create a single resource but maybe that's just because we don't have a graphql mutation for creating multiple of these at a time), but I don't have a strong preference of either one.

This is maybe more of a terraform design question since I'm not too familiar, but does it make sense to have a 'resource' in terraform refer to a number of different object (in this case multiple tags)?  Or does a resource typically have to map to a specific instance of an object?

Reply:
In Terraform's design principles: 
"If a resource in your API has an ID then it should have its own resource in terraform"

So i've been mostly following that one.

In terms of - if you just followed what we had in `resources_opslevel_service_tag.go` then lets just keep with that. 👍